parquet-column/src/main/java/org/apache/parquet/schema/PrimitiveStringifier.java

```diff
@@ -0,0 +1,358 @@
+    @Override
+    public String stringify(boolean value) {
+      return Boolean.toString(value);
+    }
+
+    @Override
+    public String stringify(int value) {
+      return Integer.toString(value);
+    }
+
+    @Override
+    public String stringify(long value) {
+      return Long.toString(value);
+    }
+
+    @Override
+    public String stringify(float value) {
+      return Float.toString(value);
+    }
+
+    @Override
+    public String stringify(double value) {
+      return Double.toString(value);
+    }
+
+    @Override
+    String stringifyNotNull(Binary value) {
+      ByteBuffer buffer = value.toByteBuffer();
+      StringBuilder builder = new StringBuilder(2 + buffer.remaining() * 2);
+      builder.append(BINARY_HEXA_PREFIX);
+      for (int i = buffer.position(), n = buffer.limit(); i < n; ++i) {
+        byte b = buffer.get(i);
+        builder.append(digits[(b >>> 4) & 0x0F]);
+        builder.append(digits[b & 0x0F]);
+      }
+      return builder.toString();
+    }
+  };
+
+  static final PrimitiveStringifier UNSIGNED_STRINGIFIER = new PrimitiveStringifier("UNSIGNED_STRINGIFIER") {
+    private static final long INT_MASK = 0x00000000FFFFFFFFl;
+
+    // Implemented based on com.google.common.primitives.UnsignedInts.toString(int, int)
+    @Override
+    public String stringify(int value) {
+      return Long.toString(value & INT_MASK);
+    }
+
+    // Implemented based on com.google.common.primitives.UnsignedLongs.toString(long, int)
+    @Override
+    public String stringify(long value) {
+      if (value == 0) {
+        // Simply return "0"
+        return "0";
+      } else if (value > 0) {
+        return Long.toString(value);
+      } else {
+        char[] buf = new char[64];
+        int i = buf.length;
+        // Split x into high-order and low-order halves.
+        // Individual digits are generated from the bottom half into which
+        // bits are moved continuously from the top half.
+        long top = value >>> 32;
+        long bot = (value & INT_MASK) + ((top % 10) << 32);
+        top /= 10;
+        while ((bot > 0) || (top > 0)) {
+          buf[--i] = Character.forDigit((int) (bot % 10), 10);
+          bot = (bot / 10) + ((top % 10) << 32);
+          top /= 10;
+        }
+        // Generate string
+        return new String(buf, i, buf.length - i);
+      }
+    }
+  };
+
+  static final PrimitiveStringifier UTF8_STRINGIFIER = new BinaryStringifierBase("UTF8_STRINGIFIER") {
+    @Override
+    String stringifyNotNull(Binary value) {
+      return value.toStringUsingUTF8();
+    }
+  };
+
+  static final PrimitiveStringifier INTERVAL_STRINGIFIER = new BinaryStringifierBase("INTERVAL_STRINGIFIER") {
+    @Override
+    String stringifyNotNull(Binary value) {
+      if (value.length() != 12) {
+        return BINARY_INVALID;
+      }
+      ByteBuffer buffer = value.toByteBuffer();
+      int pos = buffer.position();
+      String months = UNSIGNED_STRINGIFIER.stringify(buffer.getInt(pos));
```

Comment:
I don't know whether intentionally or by mistake, but [parquet-format](https://github.com/apache/parquet-format/blob/master/LogicalTypes.md#interval) specifies intervals to be little-endian. @rdblue, @julienledem, do you know whether intervals really should be little-endian?